src/main.rs

```diff
@@ -358,6 +371,50 @@ fn main() -> ExitCode {
             &args.nixpkgs_flake,
         );
         println!("{path}");
+    } else if args.subcmds.is_some() {
+        debug!("Opening manpage for {command}");
+
+        let path = get_command_path_from_cache(
+            &mut cache,
+            &entry,
+            use_channel,
+            command,
+            &args.nixpkgs_flake,
+        );
+        let mut path: PathBuf = path.into();
+        // Truncate to
+        // /nix/store/aaaaaaaaaaaaaaaaaaaaaaaaaaaa/
+        path.pop();
+        path.pop();
+        // Push to
+        // /nix/store/aaaaaaaaaaaaaaaaaaaaaaaaaaaa/share/man/man1
+        // as that is where the program manfiles reside. There are a few
+        // that are at `share/man`, but they are just improperly placed
+        path.push("share/man/man1");
+        // /nix/store/aaaaaaaaaaaaaaaaaaaaaaaaaaaa/share/man/man1/thing
+        path.push(command);
+        // /nix/store/aaaaaaaaaaaaaaaaaaaaaaaaaaaa/share/man/man1/thing.1
+        path.set_extension("1");
+        match std::fs::exists(path.as_path()) {
+            // Open the manpage
+            Ok(true) => {
+                let mut cmd = open_manpage(&path);
+                // Replace this process with the man program
+                let err = cmd.exec();
+                // This code will only run if an error occurs
+                eprintln!("{err:?}");
+                return ExitCode::FAILURE;
+            }
+            // No manpage
+            Ok(false) => {
+                eprintln!("No manpage for \"{command}\" found.");
+            }
+            // Some error
+            Err(err) => {
+                eprintln!("{err:?}");
+                return ExitCode::FAILURE;
+            }
+        }
```

Comment:
This works
`nix shell "nixpkgs#fish" --command man fish`

Idk where in nix that is made to work
`nix shell` source code https://github.com/NixOS/nix/blob/master/src/nix/env.cc

Reply:
Hmm, interesting. The nix3 command works, but then something like `nix-shell -p fish --run "man fish"` does not work.

Reply:
But you can run `nix-shell -p hello --run "man hello`, so it seems to depends on if the man page is in a seperate output or not.

Reply:
> Currently nix-shell -p package --run "man somepage" only works
occasionally and only due to a behavior of GNU's man-db that is not
documented: It tries to guess the man directory corresponding to every
entry in PATH which only works if the packages added to nix-shell have a
bin directory that gets added to PATH and is not supported by other
man page viewers like mandoc.

https://github.com/NixOS/nix/pull/4642

Reply:
So, considering that it is an undocumented feature of the default manpager, I'm not convinced we should use that logic.